### PR TITLE
fix(lsp): import map lookup for jsr subpath auto import

### PR DESCRIPTION
**Pull request by nayeemrmn**
Fixes bug reported by @lucacasonato.

### CODE REVIEW

File: tests/registry/jsr/@std/url/0.220.1/join.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
 // This module is browser compatible.
 
-import { join as posixJoin } from "jsr:/@std/path@^0.220.1/posix/join";
```
Comment (lucacasonato) [inline tests/registry/jsr/@std/url/0.220.1/join.ts:4]: Why this change?

Comment (nayeemrmn) [reply tests/registry/jsr/@std/url/0.220.1/join.ts:4]: I didn't check but I'm pretty sure this would bypass the faulty case, where tsc is using the form without the leading slash and that fails to match any import map key.

**Review by nathanwhit — APPROVED** [review]
LGTM
